src/v/finjector/stress_fiber.cc

```diff
@@ -67,12 +67,33 @@ stress_payload::stress_payload(stress_config cfg) {
     }
 }
 
-[[gnu::noinline]]
-static void spinner(int depth, int spins_per_scheduling_point) { // NOLINT
-    if (depth > 0) {
-        spinner(depth - 1, spins_per_scheduling_point);
+/**
+ * @brief Spins a given number of times at a given stack depth.
+ *
+ * The spinning function will have this many extra frames
+ * added to its stack depth using recursive calls. Each frame is one
+ * of two variants (corresponding to 0 or 1 for the template parameter).
+ * The exact series of frames is set by the seed parameter: each unique seed has
+ * a unique series of calls. This can help stress the CPU profiler by creating a
+ * large number of unique stacks.
+ * @param stack_depth
+ * @param spins_per_scheduling_point number of spins before returning
+ * @param seed the pattern of stack traces will be unique to the seed
+ */
+template<int VALUE>
+[[clang::noinline]] void spinner(
+  int stack_depth, int spins_per_scheduling_point, uint64_t seed) { // NOLINT
+    volatile int spins = 0;
+    // fmt::print("Spinner at depth {}\n", depth);
+    if (stack_depth > 0) {
+        auto to_call = seed & 1u ? spinner<1> : spinner<2>;
+        to_call(stack_depth - 1, spins_per_scheduling_point, seed >> 1u);
+        // the line below is prevent tailcall optimization would would otherwise
+        // defeat our attempt to create many stack frames (we would make the
+        // required number of calls but each call would simply replace the
+        // current stack frame, not add a new one)
```

Comment:
isn't there an attribute to disable tail call optimization?

Reply:
I don't think so, or not a reliable way. See eg here:

https://stackoverflow.com/questions/69328139/gcc-clang-disable-tail-recursion-optimization-for-single-function

which requires generic `__optimize__` attribute targeting the sibling-calls optimization and doesn't (apparently) work on clang, which is a common refrain with targeted optimization disablement on clang.

What you may be thinking of are attributes to _force_ tailcall (or fail otherwise) which are important for some cases (e.g., avoiding stack overflow when continuation passing patterns are used)?

Reply:
> What you may be thinking of are attributes to force tailcall (or fail otherwise) which are important for some cases (e.g., avoiding stack overflow when continuation passing patterns are used)?

Yeah I guess I had just assumed the inverse exists.